pkg/platform/provider/baremetal/validation/cluster.go

```diff
@@ -62,15 +67,49 @@ func ValidatClusterSpec(spec *platform.ClusterSpec, fldPath *field.Path, phase p
 }
 
 // ValidateClusterSpecVersion validates a given version.
-func ValidateClusterSpecVersion(version string, fldPath *field.Path, phase platform.ClusterPhase) field.ErrorList {
+func ValidateClusterSpecVersion(platformClient platformv1client.PlatformV1Interface, clsName, version string, fldPath *field.Path, phase platform.ClusterPhase) field.ErrorList {
 	allErrs := field.ErrorList{}
+
+	k8sValidVersions, err := getK8sValidVersions(platformClient, clsName)
+	if err != nil {
+		allErrs = append(allErrs, field.InternalError(fldPath, err))
+		return allErrs
+	}
+
 	if phase == platform.ClusterInitializing {
-		allErrs = utilvalidation.ValidateEnum(version, fldPath, spec.K8sVersions)
+		allErrs = utilvalidation.ValidateEnum(version, fldPath, k8sValidVersions)
 	}
 
 	return allErrs
 }
 
+func getK8sValidVersions(platformClient platformv1client.PlatformV1Interface, clsName string) (validVersions []string, err error) {
```

Comment:
1. upgrade.go -> getPlatformVersions   has same logic, please help consolidate
2. do we really need K8sValidVersions for backward compatibility?  do we need pack K8sValidVersions node.package in tke offline pakcage?
```
	K8sVersions          = []string{"1.18.3", "1.17.13", "1.16.9"}
	K8sVersionsWithV     = funk.Map(K8sVersions, func(s string) string {
		return "v" + s
	}).([]string)
	// K8sValidVersions for backward compatibility.
	K8sValidVersions      = append(K8sVersions, []string{"1.16.6", "1.14.10"}...)
	K8sValidVersionsWithV = funk.Map(K8sValidVersions, func(s string) string {
		return "v" + s
	}).([]string)
```


3. please notify UI update for this issue as well,  for `build` and `init`  purpose we can use `spec. K8sVersions`, for run time and UI, we should use the version from configmap